src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenRefReadonlyParameterTests.cs

```diff
@@ -0,0 +1,474 @@
+                		[in] int32& p
+                	) cil managed 
+                {
+                	.custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+                		01 00 00 00
+                	)
+                	.param [1]
+                		.custom instance void System.Runtime.CompilerServices.RequiresLocationAttribute::.ctor() = (
+                			01 00 00 00
+                		)
+                	// Method begins at RVA 0x2087
+                	// Code size 1 (0x1)
+                	.maxstack 8
+                	IL_0000: ret
+                } // end of method Program::'<<Main>$>g__local|0_0'
+            } // end of class Program
+            """);
+    }
+
+    [Theory]
+    [InlineData("var x = (ref readonly int p) => { };")]
+    [InlineData("var x = local; void local(ref readonly int p) { }")]
+    public void AnonymousDelegate(string def)
+    {
+        var source = $"""
+            {def}
+            System.Console.WriteLine(((object)x).GetType());
+            """;
+        var verifier = CompileAndVerify(source, targetFramework: TargetFramework.NetStandard20,
+            expectedOutput: "<>f__AnonymousDelegate0`1[System.Int32]");
+        verifier.VerifyDiagnostics();
+        verifier.VerifyTypeIL("<>f__AnonymousDelegate0`1", """
+            .class private auto ansi sealed '<>f__AnonymousDelegate0`1'<T1>
+                extends [netstandard]System.MulticastDelegate
+            {
+                .custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+                	01 00 00 00
+                )
+                // Methods
+                .method public hidebysig specialname rtspecialname 
+                	instance void .ctor (
+                		object 'object',
+                		native int 'method'
+                	) runtime managed 
+                {
+                } // end of method '<>f__AnonymousDelegate0`1'::.ctor
+                .method public hidebysig newslot virtual 
+                	instance void Invoke (
+                		[in] !T1& arg
+                	) runtime managed 
+                {
+                	.param [1]
+                		.custom instance void System.Runtime.CompilerServices.RequiresLocationAttribute::.ctor() = (
+                			01 00 00 00
+                		)
+                } // end of method '<>f__AnonymousDelegate0`1'::Invoke
+            } // end of class <>f__AnonymousDelegate0`1
+            """);
+    }
+
+    [Fact]
+    public void FunctionPointer()
+    {
+        var source = """
+            class C
+            {
+                public unsafe void M(delegate*<ref readonly int, void> p) { }
+            }
+            """;
+        var verifier = CompileAndVerify(source, options: TestOptions.UnsafeReleaseDll, targetFramework: TargetFramework.NetStandard20,
+            sourceSymbolValidator: verify); // PROTOTYPE:, symbolValidator: verify
```

Comment:
Could we include the validation in this PR given that it pertains to the current change?

Reply:
Currently, `ref readonly` parameters in function pointers are emitted exactly as `in` parameters, that's why `symbolValidator` doesn't work here. I'm not yet entirely sure what to do about that - perhaps add `modreq` (but then I think it could break users when changing between `in` and `ref readonly` which we try to avoid with this feature) or `modopt` (not sure how exactly would that behave with function pointers). So I was going to deal with it in later PR.

Reply:
The PR does what is currently spec'ed, namely treat `ref readonly` like `in` in a function pointer. I don't know much about what led to that decision.
I think we can capture that behavior in test.